### PR TITLE
Cut AnalyzeResponse over to Writeable

### PR DESCRIPTION
**Pull request by romseygeek**
This commit makes AnalyzeResponse and its various helper classes implement
Writeable.  The classes are also now immutable.

Relates to #34389

### CODE REVIEW

**Comment by romseygeek** [comment]
I started to look at AnalysisRequest as well, but that's going to be a bigger change, partially because it's used in the HLRC and so making things immutable will be a breaking change for clients.

File: server/src/main/java/org/elasticsearch/action/admin/indices/analyze/DetailAnalyzeResponse.java [hunk 1]
```diff
@@ -353,13 +325,10 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
-    public static class CharFilteredText implements Streamable, ToXContentObject {
+    public static class CharFilteredText implements Writeable, ToXContentObject {
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/action/admin/indices/analyze/DetailAnalyzeResponse.java]: should we make class variables `name` and `texts`  `final` ?

File: server/src/main/java/org/elasticsearch/action/admin/indices/analyze/DetailAnalyzeResponse.java [hunk 2]
```diff
@@ -251,7 +233,7 @@ public void writeTo(StreamOutput out) throws IOException {
         }
     }
 
-    public static class AnalyzeTokenList implements Streamable, ToXContentObject {
+    public static class AnalyzeTokenList implements Writeable, ToXContentObject {
```
Comment (mayya-sharipova) [inline server/src/main/java/org/elasticsearch/action/admin/indices/analyze/DetailAnalyzeResponse.java]: should we make `name` and `tokens` fields to be `final`?

File: server/src/main/java/org/elasticsearch/action/admin/indices/analyze/DetailAnalyzeResponse.java (hunk 1 above)
Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/action/admin/indices/analyze/DetailAnalyzeResponse.java]: ++

File: server/src/main/java/org/elasticsearch/action/admin/indices/analyze/DetailAnalyzeResponse.java (hunk 2 above)
Comment (romseygeek) [reply server/src/main/java/org/elasticsearch/action/admin/indices/analyze/DetailAnalyzeResponse.java]: ++

**Comment by romseygeek** [comment]
@elasticmachine run elasticsearch-ci/docbldesx
@elasticmachine run elasticsearch-ci/packaging-sample

**Review by mayya-sharipova — APPROVED** [review]
Thanks @romseygeek

File: server/src/main/java/org/elasticsearch/action/support/single/shard/TransportSingleShardAction.java [hunk 3]
```diff
@@ -118,8 +119,18 @@ protected void doRun() throws Exception {
             }
         });
     }
+
+    @Deprecated
     protected abstract Response newResponse();
 
+    protected Writeable.Reader<Response> getResponseReader() {
```
Comment (talevy) [inline server/src/main/java/org/elasticsearch/action/support/single/shard/TransportSingleShardAction.java]: thanks for this @romseygeek!
I tried my hand at a follow-up to remove the adapter for this here #41985.

**Pull request merged**